crypto/ec/ec_check.c

```diff
@@ -10,13 +10,28 @@
 #include "ec_lcl.h"
 #include <openssl/err.h>
 
+int EC_GROUP_check_named_curve(const EC_GROUP *group, int nist_only)
+{
+    int nid;
+
+    nid = ec_curve_nid_from_params(group);
+    if (nid > 0 && (!nist_only || EC_curve_nid2nist(nid) != NULL))
+        return nid;
+    return 0;
+}
+
 int EC_GROUP_check(const EC_GROUP *group, BN_CTX *ctx)
```

Comment:
Shouldn't the new function above be added somewhere in the EC check functions so that it's called when either calling EVP_PKEY_public_check, EVP_PKEY_check or EVP_PKEY_param_check?

Reply:
It could go into EVP_PKEY_param_check() for FIPS mode (when that exists) since only approved NIST named curves are allowed in that mode.  I dont know that this check should be done if custom curves are allowed (i.e non FIPS mode).